lib/model/folder.go

```diff
@@ -701,6 +722,24 @@ func (f *folder) scanTimerFired() {
 	f.Reschedule()
 }
 
+func (f *folder) versionCleanupTimerFired() {
+	f.setState(FolderCleanWaiting)
+	defer f.setState(FolderIdle)
+
+	if err := f.ioLimiter.takeWithContext(f.ctx, 1); err != nil {
+		return
+	}
+	defer f.ioLimiter.give(1)
+
+	f.setState(FolderCleaning)
+
+	if err := f.versioner.Clean(f.ctx); err != nil {
+		l.Infoln("Failed to clean versions in %s: %v", f.Description(), err)
+	}
+
+	f.versionCleanupTimer.Reset(f.versionCleanupInterval)
```

Comment:
Does a change in this trigger a folder restart?

Reply:
Looks like any folder config attribute without `restart:"false"` triggers a folder restart, and the versioning config is without attribute. So yeah, I expect any change to the versioning config should trigger a restart appropriately.